src/main/java/io/github/acm19/aws/interceptor/http/AwsRequestSigningApacheInterceptor.java

```diff
@@ -101,8 +101,15 @@ public void process(final HttpRequest request, final HttpContext context)
         requestBuilder.headers(headerArrayToMap(request.getAllHeaders()));
         SdkHttpFullRequest signedRequest = signer.signRequest(requestBuilder.build());
 
-        // copy everything back
-        request.setHeaders(mapToHeaderArray(signedRequest.headers()));
+        if (!isServerlessOpenSearch) {
+            // copy everything back
+            request.setHeaders(mapToHeaderArray(signedRequest.headers()));
+        } else {
+            // copy everything back, don't override headers as no all of them were used for signing the request
```

Comment:
no -> not

Why is this different for serverless? What headers should not be removed? Does the serverless version not work against managed service?

Reply:
TL;DR: `Content-Length`.

For Managed OpenSearch we include `Content-Length` in the signature when it's different that `0`, so it ends up in `signedRequest` so we just override all the headers and that's it. In the case of OpenSearch Sls we don't include it in the signature, if we override all headers we get rid of it and OpenSearch Sls doesn't like that. In order to preserve it then add back headers that are included in the request while keeping the original headers. It's not possible to do this for Managed OpenSearch because you'd then have it for `content-length` equals `0` which I understand we shouldn't, so test fails, but I didn't test against an actual instance, maybe I should before we merge this.